src/miner.cpp

```diff
@@ -39,13 +39,21 @@ int64_t UpdateTime(CBlockHeader* pblock, const Consensus::Params& consensusParam
     return nNewTime - nOldTime;
 }
 
-void RegenerateCommitments(CBlock& block, CBlockIndex* prev_block)
+void RegenerateCommitments(CBlock& block, ChainstateManager& chainman)
```

Comment:
In commit "miner: Pass in chainman to RegenerateCommitments" (f5368a525b3d5b93d06c83101497de5aaa6fe1c8)

It seems like this could take blockman reference instead of a chainman reference to be a little more limited. Doesn't matter though, all version of this function seem about the same to me.

Reply:
Unfortunately if we take a blockman reference, then callers would need to take `cs_main` to reach into ChainMan for the blockman ref :-/

Something for another day I think! Added to: #21766